Engine/Engine.cs

```diff
@@ -48,38 +47,32 @@ namespace QuantConnect.Lean.Engine
     public class Engine
     {
         private readonly bool _liveMode;
-        private readonly LeanEngineSystemHandlers _systemHandlers;
-        private readonly LeanEngineAlgorithmHandlers _algorithmHandlers;
-        private readonly Lazy<StackExceptionInterpreter> _exceptionInterpreter
-            = new Lazy<StackExceptionInterpreter>(() => StackExceptionInterpreter.CreateFromAssemblies(AppDomain.CurrentDomain.GetAssemblies()));
+        private readonly Lazy<StackExceptionInterpreter> _exceptionInterpreter;
 
         /// <summary>
         /// Gets the configured system handlers for this engine instance
         /// </summary>
-        public LeanEngineSystemHandlers SystemHandlers
-        {
-            get { return _systemHandlers; }
-        }
+        public LeanEngineSystemHandlers SystemHandlers { get; }
 
         /// <summary>
         /// Gets the configured algorithm handlers for this engine instance
         /// </summary>
-        public LeanEngineAlgorithmHandlers AlgorithmHandlers
-        {
-            get { return _algorithmHandlers;}
-        }
+        public LeanEngineAlgorithmHandlers AlgorithmHandlers { get; }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="Engine"/> class using the specified handlers
         /// </summary>
         /// <param name="systemHandlers">The system handlers for controlling acquisition of jobs, messaging, and api calls</param>
         /// <param name="algorithmHandlers">The algorithm handlers for managing algorithm initialization, data, results, transaction, and real time events</param>
-        /// <param name="liveMode">True when running in live mode, false otherwises</param>
+        /// <param name="liveMode">True when running in live mode, false otherwise</param>
         public Engine(LeanEngineSystemHandlers systemHandlers, LeanEngineAlgorithmHandlers algorithmHandlers, bool liveMode)
         {
             _liveMode = liveMode;
-            _systemHandlers = systemHandlers;
-            _algorithmHandlers = algorithmHandlers;
+            SystemHandlers = systemHandlers;
+            AlgorithmHandlers = algorithmHandlers;
+            // this is slow, so we will do it on demand
+            _exceptionInterpreter = new Lazy<StackExceptionInterpreter>(() =>
```

Comment:
This is definitely not a bad way of doing it. Another thought is firing off a task to force initialization of the lazy on a threadpool thread so it can be initialized by the time it's used. Now, this is likely not important, since often times we're invoking the interpreter _after_ a fatal failure, so performance becomes less of a concern is that situation (failing/about to fail).